src/librustc_mir/interpret/intrinsics.rs

```diff
@@ -239,7 +239,52 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             "transmute" => {
                 self.copy_op_transmute(args[0], dest)?;
             }
+            "simd_insert" => {
+                let index = self.read_scalar(args[1])?.to_u32()? as u64;
+                let scalar = args[2];
```

Comment:
This is quite confusing to call something of type `OpTy` a "scalar". Those should have type `Scalar`, or this should be called somewhere else, or there should at least be a comment.